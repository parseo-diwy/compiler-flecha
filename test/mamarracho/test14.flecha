
def print x =
  case x
  | Int   -> unsafePrintInt  x
  | Char  -> unsafePrintChar x
  | True  -> (unsafePrintChar 'T';
              unsafePrintChar 'r';
              unsafePrintChar 'u';
              unsafePrintChar 'e')
  | False -> (unsafePrintChar 'F';
              unsafePrintChar 'a';
              unsafePrintChar 'l';
              unsafePrintChar 's';
              unsafePrintChar 'e')
  | Closure -> (unsafePrintChar '<';
                unsafePrintChar 'f';
                unsafePrintChar 'u';
                unsafePrintChar 'n';
                unsafePrintChar '>')

def println x = print x; print '\n'

def main =
  println 42;
  print 'H';
  print 'o';
  print 'l';
  print 'a';
  print '\n';
  println (\ x y -> x);
  println True;
  println False

