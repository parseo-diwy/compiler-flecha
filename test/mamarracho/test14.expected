42
Hola
<fun>
True
False
